### PR TITLE
Show warning when local storages are not empty

### LINKED ISSUE #8: Show a warning when storage dirs are not empty
In most cases, one wants to start with fresh storages, except `INPUT.json` in default key value store. The library should print a user-friendly warning which would explain that some of the storages are not empty.

The best place to put those checks would probably be the `ensure` functions in the `index.js`.

### PR DESCRIPTION
**Pull request by AndreyBykov**
Actor would now show a warning showing storage path when the storage is being opened. Closes #8.

Few things to mention/discuss:

1. I put it in `storageClient.get()` functions, not `_ensure` functions. It was a bit misleading in the beginning and I spent quite a while figuring out this and that, and I actually changed my mind where to put it in 3 times, and ended up with current option due to:
- `_ensure` functions do not have full path, and in case there are more storages, thus there are folder on next level, it would require to check all folders recursively + user might want to have empty default, but non-empty persisted named dataset -> notification could be misleading.
- `_ensure` function are called twice by storage manager (`_getOrCreate()`, and then in `new StorageConstructor()`. This will double the logs.
- request queue would have 3 files in the folder during the first call, 1 file during the second call, so it's another problem specific to the new request queue local implementation. When `storageClient.get()` with `apify run -p` is used - it's actually empty.

2. Since this check is invoked for each storage (default and named) - I check all the storages and print the storage directory path for convenience. Could be removed, but I think it should be handy when running locally.

3. Warning text - I could also add:
- number of files in the directory (except the queue dir),
- Suggestion to use Apify CLI (not sure about both though).

Two more things:
1. Why the 3 `this.is<Store>DirInitialized` vars in `ApifyStorageLocal` (index.js) are always `false`? It is said that it's done for lazy directories creation, but since they are always `false` - `fs.ensureDirSync()` is always invoked anyway - is it intentional and I am missing something or it should not be that way?
2. Returning to my one of initial questions - fs vs globby - I actually see some scenarios when globby would be handier (like recursive folders check is much simpler and cleaner with globby), but we don't really need it here.

Otherwise - checked locally - seems everything's good now for default/named (or combination of these) storages. So waiting for comments.

### CODE REVIEW

**Review by mnmkng — CHANGES_REQUESTED** [review]
Hmm, I wrote the below comments and then read your comments and realized there's a bug in the ensure functions. You're correct that the always false `isInitialized` fields are wrong. They should be set to `true` in the `_ensure` functions. This way, everything will only be called once when the first storage of the given type is created. Sorry for the confusion. Lesson learnt: If you see some obvious crap, it's probably a bug 😄 

> I like the nicely written messages and the way you structured the code. The problem I see is the fact that this check will be run (and message printed) every time someone calls the `get` function. It also will not get printed when someone calls the `getOrCreate` function, which is basically the same and should have the same behavior.

> You can fix the problem by adding a field to the respective class instances, but what if someone creates a new client? Like `storageLocal.requestQueue('abcdef')` It will have the field empty, so the message will get logged again. So this does not work either.

> That's why I wanted to put the finding logic into the `_ensure` functions. Because it would be called only once, when you create an instance of `ApifyStorageLocal`.

> Yeah, in the `_ensure` functions, you don't have access to the folder names, but you don't need those. You can list all the folders and look inside them to check if they're empty. Except `INPUT.json` in key-value store folders.

> It's a bit more code, but a bit better implementation. I'm sure that now, given how SDK is implemented, this most likely does not log more than once. But maybe it would log when the user would use some of the utility functions like `handledCount` on a `RequestQueue`.

> Was there any issue with putting it into the `_ensure` functions?

**Comment by AndreyBykov** [comment]
Ok - this way - I guess it's ok to put it in `_ensure` functions in the end. When I tested it - I had in mind checking of separate folders explicitly, but if we could check top storage folders only:
1. If we want to check NOT ONLY default folders - this is a case when `globby` would be handy to use. It would be 1 line instead of try-catch + recursive check of folders. It does not throw if folder does not exist, you could specify which folders to use and which files to avoid in the same array (and I guess this is why it's used in CLI). Btw was surprised to know that it's the same guy as OW package - sindresorhus.
2. When you try to check folders separately - it would check only the first one (basically, how it should work) - and this is exactly what made me think in the end to put it somewhere else. But again - if this check is for top folders only - it's ok.
3. Probably last thing to mention - in `_ensure` functions - we're gonna check only the default storages, right (in this case p.1 is irrelevant, I would use fs, and check would kind of replicate the one made by CLI)? I am just thinking that it's a bit misleading if a user starts a run with named datasets (so he does not expect to have these storages cleared and see the warning).

So to sum up - if we plan to check only default folders - I will fix the `isInitialized` vars, and move the check to `_ensure` functions as you suggested originally.

**Comment by AndreyBykov** [comment]
Moved to `_ensure()` functions -> `_checkIfStorageIsEmpty()`. Tried to run it locally with default/default+named/named storages, seems ok. Tried reduce the message to 2 part (list of folders + message) instead of 3 (split message + list), second options looks nice in my opinion.

Now as we discussed it will check the contents of the storage folder -> try to find folders inside -> if there are any - will check the contents. Will skip hidden files and INPUT for key-value store, and if something else is inside -> show warning with the list of directories.

File: src/index.js [hunk 1]
```diff
@@ -164,9 +168,38 @@ class ApifyStorageLocal {
...
             this.isRequestQueueDirInitialized = true;
         }
     }
+
+    _checkIfStorageIsEmpty(storageType, storageDir) {
+        const dirsWithPreviousState = [];
+
+        const dirents = fs.readdirSync(storageDir, { withFileTypes: true });
+        for (const dirent of dirents) {
+            if (dirent.isDirectory()) {
```
Comment (mnmkng) [inline src/index.js:181]: In cases like this it's better to "return fast". This is a loop so "continue".

```js
for (const dirent of dirents) {
    if (!dirent.isDirectory()) continue;
    
    // code without nesting
}	
```

**Review by mnmkng — CHANGES_REQUESTED** [review]
Looks good. I like that you took the time to make the warning gramatically correct. Please add a test that the warnings are printed though.

File: src/index.js (hunk 1 above)
Comment (AndreyBykov) [reply src/index.js]: updated

**Comment by AndreyBykov** [comment]
Test also added - checks that there are 2 out of 3 warnings shown (for dataset and request queue, but not for Key-Value store with INPUT.json)

**Review by mnmkng — APPROVED** [review]
